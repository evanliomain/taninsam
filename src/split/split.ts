/**
 * @module string=>array
 */
/**
 * Split a string according a given separator
 * @param separator The character to use to split the string.
 * @return the function to apply on the string to split it
 * @example
 * ```
 * split('/')('a/b/cd') // ['a', 'b', 'cd']
 * ```
 * @example
 * ```
 * split('/')('') // []
 * ```
 * @example
 * ```
 * split('')('abcd') // ['a', 'b', 'c', 'd']
 * ```
 * @example
 * ```
 * split()('abcd') // ['a', 'b', 'c', 'd']
 * ```
 * @example Using the chain
 * ```
 * chain('a/b/cd')
 *   .chain(split('/'))
 *   .value() // ['a', 'b', 'cd']
 * ```
 */
export function split(
  separator: string = ''
): (str: string) => ReadonlyArray<string> {
  return (str: string) => str.split(separator).filter(s => '' !== s);
}
