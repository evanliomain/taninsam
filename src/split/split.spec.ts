import { describe, expect, test } from 'vitest';
import { split } from './split';

describe('split function', () => {
  const input = 'a/b/cd';

  test('is a pure function', () => {
    split()(input);
    expect(input).toMatchSnapshot();
  });

  test('input |> split("/") === ["a", "b", "cd"]', () => {
    expect(split('/')(input)).toEqual(['a', 'b', 'cd']);
  });

  test('input |> split("") === ["a", "/", "b", "/", "c", "d"]', () => {
    expect(split('')(input)).toEqual(['a', '/', 'b', '/', 'c', 'd']);
  });

  test('input |> split() === ["a", "/", "b", "/", "c", "d"]', () => {
    expect(split('')(input)).toEqual(['a', '/', 'b', '/', 'c', 'd']);
  });

  test('"" |> split("/") === []', () => {
    expect(split('/')('')).toEqual([]);
  });

  test('"a,s,x,,,c,,," |> split(",") === ["a", "s", "x", "c"]', () => {
    expect(split(',')('a,s,x,,,c,,,')).toEqual(['a', 's', 'x', 'c']);
  });
});
