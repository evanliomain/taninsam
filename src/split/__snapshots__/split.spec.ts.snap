// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`split function is a pure function 1`] = `"a/b/cd"`;
